gzip.go

```diff
@@ -63,19 +70,34 @@ func addLevelPool(level int) {
 // GzipResponseWriter provides an http.ResponseWriter interface, which gzips
 // bytes before writing them to the underlying response. This doesn't close the
 // writers, so don't forget to do that.
+// It can be configured to skip response smaller than minSize.
 type GzipResponseWriter struct {
 	http.ResponseWriter
-	index int // Index for gzipWriterPools.
-	gw    *gzip.Writer
+	index        int // Index for gzipWriterPools.
+	gw           *gzip.Writer
+	minSize      int       // Specifed the minimum response size to gzip. Only the first Write call is checked.
+	chosenWriter io.Writer // After the first Write call the ResponseWriter must use the same writer.
 }
 
 // Write appends data to the gzip writer.
 func (w *GzipResponseWriter) Write(b []byte) (int, error) {
+	// If a partial response have been send, the next response will use the same writer.
+	if w.chosenWriter != nil {
+		return w.chosenWriter.Write(b)
+	}
+
+	// If b is smaller than the minSize, the regular writer is used.
+	if len(b) < w.minSize {
```

Comment:
This isn't sufficient. We must keep track of the total number of bytes which have been written, because `Write` can be called many times. We can't decide whether to gzip the response or not until the headers are written; and even then, we can only make the call based on how many bytes have been written _so far_.